---
title: "Primate_data_correction_and_verification"
author: "Drew"
date: "2023-06-20"
output:
  word_document: default
  pdf_document: default
  html_document: default
editor_options:
  chunk_output_type: console
---
This is a reanalysis of 3_Linear_Model_Tests

Colleagues Caleb Ghione and Matt Dean (University of Southern California) have pointed out phyl.vcv does not look for tip names in the data matrix and matches based on order of data. All other code DOES perform this match and is therefore correct.

We further noticed our reruns were encountering errors and generating different values for 6 of our models. Upon inspection, we found all of these were originally using the free.lambda model and the model generated an estimate of lambda that was much great than 1 or less than 0. This caused some concern and we elected to restrict lambda to between 1 and 0 by excluding free.lambda as an option when it did this.

The code uses the technique I wrote from 8_rpkm_to_ARE_analysis. This way a failure of the model to converge is seen in the output and the loop continues on to the next possible model; whereas before I would just delete the option to do that model and create multiple loops.

Lastly, I wanted to compare the coefficient of the model to the correlation value as these should be the same at lamba=0 and therefore very close. To do so I scaled the data. I did check that unscaled data will result in the same outputs.

Binomial tests are missing from this version as none are significant

The phylo heat map is also missing as that one matched tip labels with the dataframe.

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

relevant packages and data

````{r}
library(phytools)
library(nlme)
library(vegan)
````

dataframes and trees

````{r}
df<-read.csv("~/S5-primdfall.csv")
tree<-read.nexus("~/treeall.nex")
treenoT<-read.nexus("~/treenoT.nex")
````

Prep the data for analysis
````{r}
###This is the new stuff!! Reorder the dataframe to match tip labels
df<-df[match(tree$tip.label,df$Species),] 
###Follow original work
Mass<-resid(lm(log(massm)~log(massf),df))
Canine<-resid(lm(log(k9m)~log(k9f),df))
Testes<-resid(lm(log(testes)~log(masst),na.action=na.exclude,df))
TestesnoT<-Testes[complete.cases(Testes)]
dtf<-data.frame(Testes,Mass,Canine,df[,8:25])
rownames(dtf)<-df$Species
hre<-prcomp(dtf[,4:21])
###HRE PCA output is the same as before. The first dimension is the most appropriate to use
PC_ARE<-hre$x[,1]
dtf<-data.frame(dtf,PC_ARE)
dtf$Species<-df$Species
rownames(dtf)<-dtf$Species
dtfnoT<-dtf[complete.cases(dtf),]
###Scale the two dataframes independently
dtfnoT<-data.frame(scale(dtfnoT[,1:22]),dtfnoT[,23])
colnames(dtfnoT)<-c(colnames(dtf[,1:22]),"Species")
dtf<-data.frame(scale(dtf[,1:22]),dtf[,23])
colnames(dtf)<-c(colnames(dtf[,1:22]),"Species")
````

Generate the Mass Data Set
This has notes on the changes discussed above
````{r}
Mass<-dtf$Mass
###notice the new term "coefficient"
mass.results<-data.frame(are=factor(),model=factor(),correlation=numeric(),pval=numeric(),r2=numeric(),lambda=numeric(),coefficient=numeric())
for(i in 3:22)
	{
	 namei<-colnames(dtf)[i]
	 v<-dtf[,i]
	 l<-phylosig(tree,resid(gls(v~Mass,dtf)),method="lambda",test=T)$lambda
	 ###keep lambda bound
	 if(l>1)
	 	  {
	 	  	l=1
	 	  }
	 if(l<0)
	 	  {
	 	  	l=0
	 	  }
	 no.lambda<-gls(v~Mass,method="ML",dtf)
	 fix.lambda<-gls(v~Mass,correlation=corPagel(l,phy=tree,form=~Species,fixed=TRUE),method="ML",dtf)
	 ####This part of the loop tests if the model fails to converge
	 ####A successful model is kept, a failed model moves onto different starting lambda values (1 and 0)
	 t1<-try(gls(v~Mass,correlation=corPagel(l,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 loadError<-(is(t1,'try-error')|is(t1,'error'))
	 if(loadError=="FALSE")
	 	{
	 	 free.lambda<-gls(v~Mass,correlation=corPagel(l,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 	 ####This part binds the lambda value between 0 and 1. Sometimes the absolute value of lambda is so large that the extraction of its value leads to a null. If any of those events happen I set free.lambda to be fix.lambda
	 	 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 	 }else
	 	{
	 	 t1<-try(gls(v~Mass,correlation=corPagel(1,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 if(loadError==F)
	 		{
	 		 free.lambda<-gls(v~Mass,correlation=corPagel(1,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 		l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 		 }else
	 		{
	 		 t1<-try(gls(v~Mass,correlation=corPagel(0,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 	 if(loadError==F)
	 			{
	 		  	 free.lambda<-gls(v~Mass,correlation=corPagel(0,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 		  	 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 			}else
	 			{
	 			 free.lambda<-fix.lambda
	 			}
	 	 	}
	 	}
	 ####Put fix lambda as first so the aov will favor it over the other two when AIC is equal
	 mod.choose<-anova(fix.lambda,free.lambda,no.lambda)
	 model<-rownames(mod.choose)[which(min(mod.choose$AIC)==mod.choose$AIC)]
	 if(length(model>1))
	  {
	   model=model[1]
	  }
	 if(model=="free.lambda")
	 	{
	 	 l<-attr(free.lambda$apVar,"Pars")[1]
	   ###This ONE LINE was the issue from the previous version.
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtf[,2:22]),vcv(tree),l)$R)[1,i-1]	
	 	 p<-summary(free.lambda)$tTable[2,4]
	 	 aov.free<-aov(free.lambda)
	 	 RSS<-summary(aov.free)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.free)[[1]]["Mass", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 ####This line extracts the coefficient
	 	 c<-free.lambda$coefficients[2]
	 	}
	 if(model=="fix.lambda")
	 	{
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtf[,2:22]),vcv(tree),l)$R)[1,i-1]	
	 	 p<-summary(fix.lambda)$tTable[2,4]
	 	 aov.fix<-aov(fix.lambda)
	 	 RSS<-summary(aov.fix)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.fix)[[1]]["Mass", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-fix.lambda$coefficients[2]
	 	}
	 if(model=="no.lambda")
	 	{
	 	 l=0
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtf[,2:22]),vcv(tree),l)$R)[1,i-1]	
	 	 p<-summary(no.lambda)$tTable[2,4]
	 	 aov.no<-aov(no.lambda)
	 	 RSS<-summary(aov.no)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.no)[[1]]["Mass", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-no.lambda$coefficients[2]
	 	}
	 dfi<-data.frame(namei,model,cor,p,r,l,c)
	 colnames(dfi)<-c("are","model","correlation","pval","r2","lambda","coefficient")
	 mass.results<-rbind(mass.results,dfi)
	}
mass.results
````

Canine now

````{r}
Canine<-dtf$Canine
canine.results<-data.frame(are=factor(),model=factor(),correlation=numeric(),pval=numeric(),r2=numeric(),lambda=numeric(),coefficient=numeric())
for(i in 4:22)
	{
	 namei<-colnames(dtf)[i]
	 v<-dtf[,i]
	 l<-phylosig(tree,resid(gls(v~Canine,dtf)),method="lambda",test=T)$lambda
	 if(l>1)
	 	  {
	 	  	l=1
	 	  }
	 if(l<0)
	 	  {
	 	  	l=0
	 	  }
	 no.lambda<-gls(v~Canine,method="ML",dtf)
	 fix.lambda<-gls(v~Canine,correlation=corPagel(l,phy=tree,form=~Species,fixed=TRUE),method="ML",dtf)
	 t1<-try(gls(v~Canine,correlation=corPagel(l,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 loadError<-(is(t1,'try-error')|is(t1,'error'))
	 if(loadError=="FALSE")
	 	{
	 	 free.lambda<-gls(v~Canine,correlation=corPagel(l,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 	l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 	 }else
	 	{
	 	 t1<-try(gls(v~Canine,correlation=corPagel(1,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 if(loadError==F)
	 		{
	 		 free.lambda<-gls(v~Canine,correlation=corPagel(1,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 		 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 		}else
	 		{
	 		 t1<-try(gls(v~Canine,correlation=corPagel(0,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf))
	 	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 	 if(loadError==F)
	 			{
	 		  	 free.lambda<-gls(v~Canine,correlation=corPagel(0,phy=tree,form=~Species,fixed=FALSE),method="ML",dtf)
	 		  	 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 			}else
	 			{
	 			 free.lambda<-fix.lambda
	 			}
	 	 	}
	 	 }
	 mod.choose<-anova(fix.lambda,free.lambda,no.lambda)
	 model<-rownames(mod.choose)[which(min(mod.choose$AIC)==mod.choose$AIC)]
	 if(length(model>1))
	  {
	   model=model[1]
	  }
	 if(model=="free.lambda")
	  {
	 	 l<-attr(free.lambda$apVar,"Pars")[1]
	   cor<-cov2cor(phyl.vcv(as.matrix(dtf[,3:22]),vcv(tree),l)$R)[1,i-2]	
	 	 p<-summary(free.lambda)$tTable[2,4]
	 	 aov.free<-aov(free.lambda)
	 	 RSS<-summary(aov.free)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.free)[[1]]["Canine", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-free.lambda$coefficients[2]
	 	}
	 if(model=="fix.lambda")
	 	{
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtf[,3:22]),vcv(tree),l)$R)[1,i-2]	
	 	 p<-summary(fix.lambda)$tTable[2,4]
	 	 aov.fix<-aov(fix.lambda)
	 	 RSS<-summary(aov.fix)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.fix)[[1]]["Canine", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-fix.lambda$coefficients[2]
	 	}
	 if(model=="no.lambda")
	 	{
	 	 l=0
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtf[,3:22]),vcv(tree),l)$R)[1,i-2]	
	 	 p<-summary(no.lambda)$tTable[2,4]
	 	 aov.no<-aov(no.lambda)
	 	 RSS<-summary(aov.no)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.no)[[1]]["Canine", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-no.lambda$coefficients[2]
	 	}
	 dfi<-data.frame(namei,model,cor,p,r,l,c)
	 colnames(dfi)<-c("are","model","correlation","pval","r2","lambda","coefficient")
	 canine.results<-rbind(canine.results,dfi)
	 }
canine.results
````

Testes now

````{r}
TestesnoT<-dtfnoT$Testes
testes.results<-data.frame(are=factor(),model=factor(),correlation=numeric(),pval=numeric(),r2=numeric(),lambda=numeric(),coefficient=numeric())
for(i in 2:22)
	{
	 namei<-colnames(dtfnoT)[i]
	 v<-dtfnoT[,i]
	 l<-phylosig(treenoT,resid(gls(v~Testes,dtfnoT)),method="lambda",test=T)$lambda
	  if(l>1)
	 	  {
	 	  	l=1
	 	  }
	 if(l<0)
	 	  {
	 	  	l=0
	 	  }
	 no.lambda<-gls(v~TestesnoT,method="ML",dtfnoT)
	 fix.lambda<-gls(v~TestesnoT,correlation=corPagel(l,phy=treenoT,form=~Species,fixed=TRUE),method="ML",dtfnoT)
	 t1<-try(gls(v~TestesnoT,correlation=corPagel(l,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT))
	 loadError<-(is(t1,'try-error')|is(t1,'error'))
	 if(loadError=="FALSE")
	 	{
	 	 free.lambda<-gls(v~TestesnoT,correlation=corPagel(l,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT)
	 	 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 	}else
	 	{
	 	 t1<-try(gls(v~TestesnoT,correlation=corPagel(1,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT))
	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 if(loadError==F)
	 		{
	 		 free.lambda<-gls(v~TestesnoT,correlation=corPagel(1,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT)
	 		 l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 		}else
	 		{
	 		 t1<-try(gls(v~TestesnoT,correlation=corPagel(0,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT))
	 	 	 loadError<-(is(t1,'try-error')|is(t1,'error'))
		 	 if(loadError==F)
	 			{
	 		  	 free.lambda<-gls(v~TestesnoT,correlation=corPagel(0,phy=treenoT,form=~Species,fixed=FALSE),method="ML",dtfnoT)
	 		    l2<-attr(free.lambda$apVar,"Pars")[1]
	 	       if(is.null(l2))
	 	      {
	 	        free.lambda<-fix.lambda
	 	      }else
	 	      {
	 	      if(l2>1)
	 	        {
	 	  	     free.lambda<-fix.lambda
	 	        }
	 	      if(l2<0)
	 	       {
	 	  	    free.lambda<-fix.lambda
	 	      }
	 	      } 
	 			}else
	 			{
	 			 free.lambda<-fix.lambda
	 			}
	 	 	}
	 	 }
	 mod.choose<-anova(fix.lambda,free.lambda,no.lambda)
	 model<-rownames(mod.choose)[which(min(mod.choose$AIC)==mod.choose$AIC)]
	 if(length(model>1))
	  {
	   model=model[1]
	  }
	 if(model=="free.lambda")
	 	{
	 	 l<-attr(free.lambda$apVar,"Pars")[1]
	   cor<-cov2cor(phyl.vcv(as.matrix(dtfnoT[,1:22]),vcv(treenoT),l)$R)[1,i]	
	 	 p<-summary(free.lambda)$tTable[2,4]
	 	 aov.free<-aov(free.lambda)
	 	 RSS<-summary(aov.free)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.free)[[1]]["TestesnoT", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-free.lambda$coefficients[2]
	 	}
	 if(model=="fix.lambda")
	 	{
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtfnoT[,1:22]),vcv(treenoT),l)$R)[1,i]	
	 	 p<-summary(fix.lambda)$tTable[2,4]
	 	 aov.fix<-aov(fix.lambda)
	 	 RSS<-summary(aov.fix)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.fix)[[1]]["TestesnoT", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-fix.lambda$coefficients[2]
	 	}
	 if(model=="no.lambda")
	 	{
	 	 l=0
	 	 cor<-cov2cor(phyl.vcv(as.matrix(dtfnoT[,1:22]),vcv(treenoT),l)$R)[1,i]	
	 	 p<-summary(no.lambda)$tTable[2,4]
	 	 aov.no<-aov(no.lambda)
	 	 RSS<-summary(aov.no)[[1]]["Residuals", "Sum Sq"]
	 	 TSS<-summary(aov.no)[[1]]["TestesnoT", "Sum Sq"]+RSS
	 	 r<-1-RSS/TSS
	 	 c<-no.lambda$coefficients[2]
	 	}
	 dfi<-data.frame(namei,model,cor,p,r,l,c)
	 colnames(dfi)<-c("are","model","correlation","pval","r2","lambda","coefficient")
	 testes.results<-rbind(testes.results,dfi)
}
testes.results
````

reimport the data to use unscaled version
this is for analysis using canonical correspondence in vegan

````{r}
df<-read.csv("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/S5-primdfall.csv")
treenoT<-read.nexus("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/treenoT.nex")

###Vegan does not use phylogeny so no need to match tip labels... Using older code with labels already designed for given data
###Follow original work
Mass<-resid(lm(log(massm)~log(massf),df))
Canine<-resid(lm(log(k9m)~log(k9f),df))
Testes<-resid(lm(log(testes)~log(masst),na.action=na.exclude,df))
TestesnoT<-Testes[complete.cases(Testes)]
dtf<-data.frame(Testes,Mass,Canine,df[,8:25])
rownames(dtf)<-df$Species
hre<-prcomp(dtf[,4:21])
###HRE PCA output is the same as before. The first dimension is the most appropriate to use
PC_ARE<-hre$x[,1]
dtf<-data.frame(dtf,PC_ARE)
dtf$Species<-df$Species
rownames(dtf)<-dtf$Species
dtfnoT<-dtf[complete.cases(dtf),]
````

split data frame into hres and physical traits

````{r}
hre.set<-dtfnoT[,4:21]
trait.set<-dtfnoT[,1:3]
````

Using CCorA

````{r}
veg.cca<-cca(hre.set~.,data=trait.set)

##find marginal effects

anova(veg.cca, by= 'margin', permutations = 10000)

##find significant CCA

anova(veg.cca, by= 'axis', permutations = 10000)

##find significant terms

anova(veg.cca, by= 'terms', permutations = 10000)

##investigate full CCA

summary(veg.cca)

````

ploting of CCorA

````{r}
library(shape)
spcol<-c("deeppink2","deeppink2","deeppink2","darkgoldenrod3","darkgoldenrod3","darkgoldenrod3","cyan3","cyan3","cyan3","darkgoldenrod3","darkgoldenrod3","darkgoldenrod3","darkgoldenrod3","darkgoldenrod3","darkgoldenrod3","cyan3","cyan3","cyan3","darkgoldenrod3","cyan3","deeppink2","darkgoldenrod3","darkgoldenrod3")
sigsiz<-c(1.25,1.25,1.25,1.25,1.25,2,2,2,1.25,1.75,1.25,1.25,1.25,1.25,1.75,1.25,1.25,1.25)
sigcol<-c("black","black","black","black","black","blue","blue","blue","black","red","black","black","black","black","red","black","black","black")
setEPS()
postscript("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/CCorrelation.eps")
par(mfrow=c(2,2),mar=c(4,4,.5,.5))
plot(veg.cca$CCA$v[,2:1],type="n",xlim=c(-1,1),ylim=c(-1,1),cex.lab=1.25,cex.axis=1)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.cca$CCA$u),y=(veg.cca$CCA$u[,1])*.4,x=(veg.cca$CCA$u[,2])*.4,cex=0.5,col=spcol)
text(rownames(veg.cca$CCA$v),y=(veg.cca$CCA$v[,1])*.28,x=(veg.cca$CCA$v[,2])*.28,cex=sigsiz*.6,col=sigcol)
text("Canine",x=0.5,y=.86,cex=1.1)
text("Testis",x=0.6,y=.50,cex=1.1)
text("Mass",x=.93,y=-0.05,cex=1.1)
Arrows(0,0,veg.cca$CCA$biplot[2,2],veg.cca$CCA$biplot[2,1],lwd=1.25)
Arrows(0,0,veg.cca$CCA$biplot[1,2],veg.cca$CCA$biplot[1,1],lwd=1.25)
Arrows(0,0,veg.cca$CCA$biplot[3,2],veg.cca$CCA$biplot[3,1],lwd=1.25)
text("A",x=-.95,y=.95,cex=2.5)
plot(veg.cca$CCA$v[,c(3,1)],type="n",xlim=c(-1,1),ylim=c(-1,1),cex.lab=1.25,cex.axis=1)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.cca$CCA$u),y=(veg.cca$CCA$u[,1])*.4,x=(veg.cca$CCA$u[,3])*.4,cex=0.5,col=spcol)
text(rownames(veg.cca$CCA$v),y=(veg.cca$CCA$v[,1])*.28,x=(veg.cca$CCA$v[,3])*.28,cex=sigsiz*.6,col=sigcol)
text("Canine",x=0.41,y=.88,cex=1.1)
text("Testis",x=-.71,y=.52,cex=1.1)
text("Mass",x=.57,y=-.02,cex=1.1)
Arrows(0,0,veg.cca$CCA$biplot[2,3],veg.cca$CCA$biplot[2,1],lwd=1.25)
Arrows(0,0,veg.cca$CCA$biplot[1,3],veg.cca$CCA$biplot[1,1],lwd=1.25)
Arrows(0,0,veg.cca$CCA$biplot[3,3],veg.cca$CCA$biplot[3,1],lwd=1.25)
text("B",x=-.95,y=.95,cex=2.5)
dev.off()
````

````{r}
cca.phy<-phyl.cca(treenoT,trait.set,hre.set,fixed=FALSE)
cca.phy

## Significance tests indicates all three axes explain variation, but axes 2 and 3 cannot on their own
##noticed lambda is close to zero and many of ARE do not have phylogenic correlations independently

##use vegan package to do correlation analysis for easier time manipulating data

veg.ccora<-CCorA(hre.set,trait.set, stand.X = TRUE, stand.Y = TRUE, permutations = 10000)
veg.ccora
````

Plot the CCA
````{r}
setEPS()
postscript("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/CCorrespondence.eps")
par(mfrow=c(2,2),mar=c(4,5,0.5,0.5))
plot(veg.ccora$corr.X.Cx[,1],veg.ccora$corr.X.Cx[,2],col="white",xlab="CCorA1",ylab="CCorA2",xlim=c(-1,1),ylim=c(1,-1),cex.axis=1,cex.lab=1.25)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.ccora$Cx),x=veg.ccora$Cx[,1]*.4,y=veg.ccora$Cx[,2]*.4, cex = 0.5, col = spcol)
text("A",x=-.95,y=-.95,cex=2.5)
Arrows(x0=0,y0=0,x1=veg.ccora$corr.X.Cx[,1],y1=veg.ccora$corr.X.Cx[,2],lwd=1.25)
text(rownames(veg.ccora$corr.X.Cx),x=veg.ccora$corr.X.Cx[,1],y=veg.ccora$corr.X.Cx[,2]-0.08,cex =1.1)
plot(veg.ccora$corr.Y.Cy[,1],veg.ccora$corr.Y.Cy[,2],col="white",xlab="CCorA1",ylab="CCorA2",xlim=c(-1,1),ylim=c(1,-1),cex.axis=1,cex.lab=1.25)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.ccora$Cy),x=veg.ccora$Cy[,1]*0.4,y=veg.ccora$Cy[,2]*0.4, cex = 0.5, col = spcol)
text(rownames(veg.ccora$corr.Y.Cy),x=veg.ccora$corr.Y.Cy[,1]*1.5,y=veg.ccora$corr.Y.Cy[,2]*1.5, cex = sigsiz*.6, col = sigcol)
text("B",x=-0.95,y=-.95,cex=2.5)
plot(veg.ccora$corr.X.Cx[,3],veg.ccora$corr.X.Cx[,2],col="white",xlab="CCorA3",ylab="CCorA2",xlim=c(-1,1),ylim=c(1,-1),cex.axis=1,cex.lab=1.25)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.ccora$Cx),x=veg.ccora$Cx[,3]*.4,y=veg.ccora$Cx[,2]*.4, cex = 0.5, col = spcol)
Arrows(x0=0,y0=0,x1=veg.ccora$corr.X.Cx[,3],y1=veg.ccora$corr.X.Cx[,2],lwd=1.25)
text(rownames(veg.ccora$corr.X.Cx),x=veg.ccora$corr.X.Cx[,3],y=veg.ccora$corr.X.Cx[,2]-0.08,cex =1.1)
text("C",x=-0.95,y=-.95,cex=2.5)
plot(veg.ccora$corr.Y.Cy[,2],veg.ccora$corr.Y.Cy[,3],col="white",xlab="CCorA3",ylab="CCorA2",xlim=c(-1,1),ylim=c(1,-1),cex.axis=1,cex.lab=1.25)
abline(h=0,lty=2,col="gray65")
abline(v=0,lty=2,col="gray65")
text(rownames(veg.ccora$Cy),x=veg.ccora$Cy[,3]*.4,y=veg.ccora$Cy[,2]*0.4, cex = 0.5, col = spcol)
text(rownames(veg.ccora$corr.Y.Cy),x=veg.ccora$corr.Y.Cy[,3]*1.5,y=veg.ccora$corr.Y.Cy[,2]*1.5, cex = sigsiz*.6, col = sigcol)
text("D",x=-0.95,y=-.95,cex=2.5)
dev.off()
````

Redo the network figure to include the new significant results

````{r}
library(pegas)
motif<-read.dna("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/motifs.fasta",format = "fasta")
mot<-haplotype(motif)
rownames(mot)<-rownames(motif)
motnet<-haploNet(mot)
back<-c("dodgerblue","gray72","red","gray72","gray72","gray72","red","gray72","gray72","gray72","gray72","gray72","dodgerblue","dodgerblue","gray72","gray72","gray72","gray72")
fon.val=c(3,3,3,2,2,3,3,2,2,3,3,3,2,3,3,3,2,3)
wid<-c(5,5,5,5,5,5,3,3,3,3,3,3,1,1,1,1,0.5)
setEPS()
postscript("/Users/andersond/Desktop/My Pubs/Primate/Evolution/resubwork/Haplonetwork.eps")
par(mfrow=c(1,1),mar=c(1,1,1,1))
plot(motnet,size=2.5,show.mutation=1,fast=FALSE,threshold=c(0,1),scale.ratio=1,cex=.8,lty=1,lwd=wid,bg=back,font=fon.val)
dev.off()
````

Cluster analysis for CCA hre values (all three dimensions)
````{r}
library(NbClust)
library(mclust)
nb <- NbClust(as.matrix(veg.cca$CCA$v[,1:3]), diss=NULL, distance = "euclidean", min.nc=2, max.nc=10, method = "kmeans", index = "all", alphaBeale = 0.1)
wss <- sapply(1:16, function(k){kmeans(veg.cca$CCA$v[,1:3], k, nstart=50,iter.max = 16 )$tot.withinss})
wss
d_clust <- Mclust(as.matrix(veg.cca$CCA$v[,1:3]), G=1:8, modelNames = mclust.options("emModelNames"))
d_clust$BIC
````

5 or 6 looks like the best
````{r}
k.cca.hre5<-kmeans(veg.cca$CCA$v[,1:3],centers = 5, iter.max = 1000, nstart = 25)
k.cca.hre5
k.cca.hre6<-kmeans(veg.cca$CCA$v[,1:3],centers = 6, iter.max = 1000, nstart = 25)
k.cca.hre6
````

Cluster analysis for CCA species values (all three dimensions)
````{r}
nb <- NbClust(as.matrix(veg.cca$CCA$u[,1:3]), diss=NULL, distance = "euclidean", min.nc=2, max.nc=10, method = "kmeans", index = "all", alphaBeale = 0.1)
wss <- sapply(1:16, function(k){kmeans(veg.cca$CCA$u[,1:3], k, nstart=50,iter.max = 16 )$tot.withinss})
wss
d_clust <- Mclust(as.matrix(veg.cca$CCA$u[,1:3]), G=1:8, modelNames = mclust.options("emModelNames"))
d_clust$BIC
````

3 or 6 looks like the best
````{r}
k.cca.hre3<-kmeans(veg.cca$CCA$u[,1:3],centers = 3, iter.max = 1000, nstart = 25)
k.cca.hre3
k.cca.hre6<-kmeans(veg.cca$CCA$u[,1:3],centers = 6, iter.max = 1000, nstart = 25)
k.cca.hre6
````

Do the same for CCorA
```{r}
nb <- NbClust(as.matrix(veg.ccora$corr.Y.Cx[,1:3]), diss=NULL, distance = "euclidean", min.nc=2, max.nc=10, method = "kmeans", index = "all", alphaBeale = 0.1)
wss <- sapply(1:16, function(k){kmeans(veg.ccora$corr.Y.Cx[,1:3], k, nstart=50,iter.max = 16 )$tot.withinss})
wss
d_clust <- Mclust(as.matrix(veg.ccora$corr.Y.Cx[,1:3]), G=1:8, modelNames = mclust.options("emModelNames"))
d_clust$BIC
````

````{r}
k.ccora.hre<-kmeans(veg.ccora$corr.Y.Cx[,1:3],centers=5,iter.max=1000,nstart=25)
k.ccora.hre
````

````{r}
nb <- NbClust(as.matrix(veg.ccora$Cx[,1:3]), diss=NULL, distance = "euclidean", min.nc=2, max.nc=10, method = "kmeans", index = "all", alphaBeale = 0.1)
wss <- sapply(1:16, function(k){kmeans(veg.ccora$Cx[,1:3], k, nstart=50,iter.max = 16 )$tot.withinss})
wss
d_clust <- Mclust(as.matrix(veg.ccora$Cx[,1:3]), G=1:8, modelNames = mclust.options("emModelNames"))
d_clust$BIC
````


````{r}
k.ccora.sp3<-kmeans(veg.ccora$Cx[,1:3],centers=3,iter.max=1000,nstart=25)
k.ccora.sp3
k.ccora.sp6<-kmeans(veg.ccora$Cx[,1:3],centers=6,iter.max=1000,nstart=25)
k.ccora.sp6
````
